src/service/data_products/taxa_count.py

```diff
@@ -234,9 +243,43 @@ async def get_taxa_counts(
         await _add_subset_data_in_place(q, store, collection_id, load_ver, rank, dp_proc)
         # For now always sort by the std data. See if ppl want sort by match/selection data
         # before implementing something more sophisticated.
+    _sort_taxa_counts(q)
+
     return _taxa_counts(dp_match=dp_match, dp_sel=dp_sel, data=q)
 
 
+def _sort_dict_list(
+        dict_list: list[dict],
+        key: str,
+        reverse: bool = True):
+    # Sort taxa count records, a list of dicts, in place by the key of dictionary with None values last.
+    # Key values should be either numeric or None.
+    # Certain keys, such as 'sel_count' or 'match_count,' may not exist in all dictionaries.
+    # For instance, these keys might be absent if the corresponding match or selection process did not occur/complete.
+    # In such cases, the default value of float('-inf') is used to ensure that the dictionaries are sorted correctly.
+    dict_list.sort(key=lambda x: (int(x.get(key)) if x.get(key) is not None else float('-inf')), reverse=reverse)
+
+
+def _fill_missing_orders(sort_order: list[str]):
+    # fill in missing orders with the default sort order
+    if not isinstance(sort_order, list):
+        raise ValueError(f"sort_order must be a list of strings, provided: {sort_order}")
+
+    return [order for order in _DEFAULT_SORT_ORDER if order not in sort_order] + sort_order
```

Comment:
So this means, IIUC, that if the sort order I provide is "match_count,sel_count" then the sort order will be std->match->sel? That seems wrong, I'd expect match->sel->std.

Reply:
`sort order` is the order we sort. I think that's correct. If you provide ['match_count', 'sel_cout'], the sorting function will sort 'match_count' first, then 'sel_cout'. I am planning to introduce `sort_priority` as endpoint input. And something making `sort_order = sort_priority[::-1]`. Thus, if user inputs `['matched', 'selected']`, the sort_order is assigned to `['count', 'sel_count', 'match_count']`.

Reply:
Ok, I wasn't thinking this through correctly. This function is returning the *reverse* of the users's desired sort so that the last sort is the user's first priority.

Reply:
That could use a comment in the code I think

Reply:
Hold on again - shouldn't sort_order be reversed in that case?

Reply:
And _DEFAULT_SORT_ORDER for that matter

Reply:
No. I was planning to call this func with reverse of user’s priority.

Reply:
Then at  _DEFAULT_SORT_ORDER needs to be reversed too, right? Otherwise the default sort order is really match->sel->count

Reply:
No. I don't think so. Why?

Reply:
Ok, I finally grok this. I managed to confuse myself on part of it, but I think there are two confusing parts about the implementation:

* I would reverse the user's sort order inside the function rather than in the calling function, unless there a reason that it's important it's in the calling function
* Sorts happen on the match and selection fields even if there's no data to sort, so the sort is effectively a no op and a waste of cycles. I don't think the latter is significant but the next person that comes along might, like me, go "why the heck is match in the sort order when no match was provided?"  I would write things so that the sort order only contains, and therefore sorts only occur on, data that is actually in the data structure

Reply:
👍 updated func to take sort_priority and only sort on existing processes.

Reply:
:+1: